lib/App/Sqitch/Command/revert.pm

```diff
@@ -145,28 +144,17 @@ sub execute {
         ? $engine->planned_deployed_common_ancestor_id
         : $self->to_change // shift @{ $changes };
 
-    if (!defined $change && $self->strict) {
-        hurl {
-            ident   => 'revert:strict',
-            message => __ 'Must specify a target revision in strict mode',
-            exitval => 1,
-        };
-    }
+    # Require a change to revert to in strict mode.
+    hurl {
+        ident   => 'revert:strict',
+        message => __ 'Must specify a target revision in strict mode',
+        exitval => 1,
+    } if !defined $change && $self->strict;
 
-    if (@{ $changes }) {
-        if ($self->strict) {
-            hurl {
-                ident   => 'revert:strict:excess',
-                message => __ 'Too many changes specified',
-                exitval => 1,
-            };
-        }
-
-        $self->warn(__x(
-            'Too many changes specified; reverting to "{change}"',
-            change => $change,
-        ));
-    }
+    $self->warn(__x(
+        'Too many changes specified; reverting to "{change}"',
+        change => $change,
+    )) if @{ $changes };
```

Comment:
I think strict mode should make it a fatal error to specify too many changes, do you disagree?

My rationale is that in strict mode, when reverting the user should be specific about what changes should be reverted, and ignoring a user-specified change goes against that principle.

Reply:
Oh yeah, I was thinking it was fatal, but it's a warning. Will need to add a test for that.

Reply:
Fixed in e439eb79.